### ISSUE TITLE
Code fix for CA1820 could be improved for `string.Equals`

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5202]
### Analyzer

**Diagnostic ID**: [CA1820](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1820)

### Describe the improvement

In the case of an empty string using `str.Equals(string.Empty)`, it can be assumed that `str` is not null, as this would have resulted in a `NullReferenceException` in the first instance. 

#### Current code fix

```csharp
string.IsNullOrEmpty(str)
```

#### Expected code fix

```csharp
str.Length == 0
```

### Additional context

#4537

**Comment by mavasani** [comment]
Sounds reasonable!

**Comment by xtqqczze** [comment]
@hrkeni Are you interested in taking this to follow-up #4622?

**Comment by hrkeni** [comment]
@xtqqczze Yeah, happily!

**Comment by hrkeni** [comment]
@xtqqczze @mavasani @Evangelink 
Would the preferred code fix for `!str.Equals(string.Empty)` be `str.Length != 0` in this case? That would mean modifying the diagnostic to also start considering the unary negation operator as part of the fix to enable that. I'm not sure if that can have some unintended consequences or not.

Alternatively, `!str.Equals(string.Empty)` can be easily fixed `!(str.Length == 0)` but I'm not sure how I could avoid also always including the parentheses in the case without negation i.e. `str.Equals(string.Empty)` getting fixed to `(str.Length == 0)`.

I might be missing something simple here.

**Comment by xtqqczze** [comment]
> Would the preferred code fix for `!str.Equals(string.Empty)` be `str.Length != 0` in this case? 

Yes, this is preferred; I've updated the issue description to cover the negated case.

**Comment by xtqqczze** [comment]
I've also added the negated case with parenthesis to the description.

**Comment by xtqqczze** [comment]
I'm surprised there is no code style rule that covers `!(i == 0)` ➜  `i != 0`.
